### DISCUSSION TITLE
Don't know where to start

### DISCUSSION THREAD

**Discussion by hagepat** [andesse/hue-clip-api.node-red-flows#29] [Q&A] [answered]
Hi Andreas, 
First of all kudos for this flow, it looks really nice. But I have a problem reading all this stuff.

The bridge is connected successful, and I can see that there is response in the debug window when I turn on the light in the HUE app.
I'm fairly new to node red and don't understand what I have to do with the sub flow???
I have 5 HUE lights and 2 groups, do you have an example how I can use the light in the flow?
Previous I had a flow for all the lights, and in that flow I add the node from Hue Magic. 
When I read the readme the part I simply don't understand is:
- Do it room-by-room (what do you mean by that??)
- Build the whole HUE-setup in a single Flow-Tab, use link-in / link-out nodes to connect it to your Rooms (Never used Link-in / link-out)
- Label those link-in / link-out nodes, otherwise it will be too confusing (same as above)
- Trigger scenes and not single lights, too much requests at once will be refused by the bridge The Bridge is just able to receive max! 5 light / 1 group request per second, less is better. If problems do occure, use the alternative request flow (step3) (I don't have scenes for my lights)
- additional flows are in the "flows" folder in this repository and can be downloaded there (Found them but why do I need them?)
- seperate flows can be found in the "flows" subfolder (same as above?)

Attached is the screenshot I have so far, (It's in dutch, sorry for that :-))
![SNAG-0000 13-Sep-22 9 37 08 PM](https://user-images.githubusercontent.com/86970871/189993883-7eb9b049-eeae-4c0e-81b7-91533f221a2d.png)

![image](https://user-images.githubusercontent.com/86970871/189993977-7445180e-2022-49ee-b56e-cf5e9e9a440b.png)
So the question is how can I trigger the light with this new setup.

Hope that you can help me because this is a bit frustrating on my side :-)

Thanks in advance.

Kind Regards,
Patrick

**Comment by andesse** [comment]
@hagepat Hey Patrick, thanks for the compliment, I think all the users who participated will take it ;)
I’m from Germany and I can read / tell Dutch a bit :)

And what I am shocked about is, that on your Screenshots are just Huemagic nodes. :D
I can’t see a single node from my flow, explain it please. 

The subflows are for lamp / group responses. If you add the Id/rid there you’ll get a boolean true/false out instead of a big array object, just try it out. 

in my flow you’ll find some yellow change nodes with examples. Just add a lamp Id in one, inject and the light will do what is written on the node. 

When you previously just had one flow tab, than everything is fine. Lots of people have quite a lot of tabs for Livingroom, bedroom, kitchen, etc. my recommendation is to have all hue stuff in the hue flow and link everything in and out. 

- Do it room-by-room (what do you mean by that??)
- Build the whole HUE-setup in a single Flow-Tab, use link-in / link-out nodes to connect it to your Rooms (Never used Link-in / link-out) Label those link-in / link-out nodes, otherwise it will be too confusing (same as above)

What I mean with that is, that you should setup you rooms one after the other. Put all HUE nodes into one flow tab and link the rooms in this tab / and out to this tab. In detail you have a call action Change Node for your kitchen in the separate hue flow tab and should connect a link out from the kitchen switch, to the call into the hue tab. 

- Trigger scenes and not single lights, too much requests at once will be refused by the bridge The Bridge is just able to receive max! 5 light / 1 group request per second, less is better. If problems do occure, use the alternative request flow (step3) (I don't have scenes for my lights)

If you have couple of light in one room and you want to turn the on all at the same time, create a scene. 
if every lamp has an individual button, you can use lights. 

- additional flows are in the "flows" folder in this repository and can be downloaded there (Found them but why do I need them?) Separate flows can be found in the "flows" subfolder (same as above?)
 
These are files from previous releases or specials like blink a lamp as alarm and are stored there cause, as you have figured already out, you don’t really need them, it’s a collection. 

Did this help?

Greetings 
Andreas

**Reply by hagepat** [reply]
Hi Andreas, thanks for your quick reply
Sorry voor my noobisch questions. Bu can you share a screenshot how I integrate your flow to switch on/off my lamps?
Or a simple flow to switch a light on and share the flow?
I have to create subflow for every lamp, correct? , Do I have to duplicate the subflow? 
Or to I have to place a "Hue Light Receiver" for every lamp?
![image](https://user-images.githubusercontent.com/86970871/190072324-ea8ea7a4-15f6-4d6f-9b10-41020b70ad42.png)
Here is the "HUE Light Receiver" with an lamp ID. 

![image](https://user-images.githubusercontent.com/86970871/190072416-a65dce52-99d5-484e-8075-1bad0eb4652e.png)
Do I have to fill in Lamp ID in the subflow as well?

Again sorry to bother. But I want to use your flow in my lamp  schema but I simply can't figure out how it works :-)

Thanks again in advance for your answer.

Greetz Patrick

**Comment by andesse** [comment]
@hagepat yo Patrick, it’s super easy. When you figured out how it works it’s fun.

yes, you need to have a blue light resource subflow for every light or group you want to have a response from.. dublicate and ad the correct Id.

i made you a Flow with 3 nodes, blue subflow and two yellow changes node. This is a setup for ONE lamp. 
Import my example, have a look, copy the on/off nodes maybe and add these to the imported main flow..
Connect everything to the right place. (I made comments as well) add your ID inside, it’s the same Id for all three nodes. 

`[{"id":"a481bf787a7fdeab","type":"subflow","name":"HUE Light Receiver","info":"","category":"HUE v2","in":[{"x":40,"y":580,"wires":[{"id":"f3271feba7c9915a"}]}],"out":[{"x":1020,"y":500,"wires":[{"id":"7589246b1e06008f","port":0}]},{"x":1040,"y":580,"wires":[{"id":"7589246b1e06008f","port":0},{"id":"4af07ff82cd530da","port":0}]},{"x":1020,"y":660,"wires":[{"id":"4af07ff82cd530da","port":0}]}],"env":[{"name":"light_ID","type":"str","value":"replace with discovered Lamp / Group ID"}],"meta":{},"color":"#86E3FF","outputLabels":["On","On / Off","Off"],"icon":"node-red/light.svg"},{"id":"8b5bf9aa6165443b","type":"rbe","z":"a481bf787a7fdeab","name":"","func":"rbe","gap":"","start":"","inout":"out","septopics":true,"property":"payload","topi":"topic","x":730,"y":500,"wires":[["7589246b1e06008f"]]},{"id":"7589246b1e06008f","type":"switch","z":"a481bf787a7fdeab","name":"","property":"payload","propertyType":"msg","rules":[{"t":"true"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":500,"wires":[[]]},{"id":"a7ba6a6bbe78ae39","type":"change","z":"a481bf787a7fdeab","name":"false","rules":[{"t":"set","p":"payload","pt":"msg","to":"false","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":730,"y":460,"wires":[["099f85bc73efff73"]]},{"id":"099f85bc73efff73","type":"delay","z":"a481bf787a7fdeab","name":"2s","pauseType":"delay","timeout":"2","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"outputs":1,"x":850,"y":460,"wires":[["8b5bf9aa6165443b"]]},{"id":"a003a94d2ebb1a69","type":"change","z":"a481bf787a7fdeab","name":"True","rules":[{"t":"set","p":"payload","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":730,"y":540,"wires":[["8b5bf9aa6165443b"]]},{"id":"b785268eac1e03d1","type":"delay","z":"a481bf787a7fdeab","name":"2s","pauseType":"delay","timeout":"2","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":850,"y":700,"wires":[["b836e2a7909c55e9"]]},{"id":"b836e2a7909c55e9","type":"rbe","z":"a481bf787a7fdeab","name":"","func":"rbe","gap":"","start":"","inout":"out","septopics":true,"property":"payload","x":730,"y":660,"wires":[["4af07ff82cd530da"]]},{"id":"598dbd38880b8bb8","type":"change","z":"a481bf787a7fdeab","name":"false","rules":[{"t":"set","p":"payload","pt":"msg","to":"false","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":730,"y":620,"wires":[["b836e2a7909c55e9"]]},{"id":"c3b7a3bc0f18b044","type":"change","z":"a481bf787a7fdeab","name":"true","rules":[{"t":"set","p":"payload","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":730,"y":700,"wires":[["b785268eac1e03d1"]]},{"id":"4af07ff82cd530da","type":"switch","z":"a481bf787a7fdeab","name":"","property":"payload","propertyType":"msg","rules":[{"t":"false"}],"checkall":"true","repair":false,"outputs":1,"x":850,"y":660,"wires":[[]]},{"id":"db4b92b38d2635c5","type":"switch","z":"a481bf787a7fdeab","name":"payload.data.0.on.on","property":"payload.data.0.on.on","propertyType":"msg","rules":[{"t":"true"},{"t":"false"}],"checkall":"true","repair":false,"outputs":2,"x":480,"y":580,"wires":[["a003a94d2ebb1a69","a7ba6a6bbe78ae39"],["598dbd38880b8bb8","c3b7a3bc0f18b044"]]},{"id":"b94463b527fd5453","type":"comment","z":"a481bf787a7fdeab","name":"HUE light / grouped_light Event Filter \\n  \\n Output 1: This output will send a **true** when Lights or Groups are turned on. \\n  \\n Output 2: This output will send a **true** when Lights or Groups are turned on AND a **false** when Lights or Groups are turned off. \\n  \\n Output 3: This output will send a **false** when Lights or Groups are turned off \\n  \\n Depending on your Situation it can be useful just to have a **true** OR **false** OR **both** states in your Flow. Thats the reason i built this Subflow. \\n Just use the Output that makes you happy :) \\n  \\n ----------------------- \\n All outputs are boolean  \\n ----------------------- \\n  \\n The delays will reset the RBE nodes. \\n You can adjust the times for your needs, but this worked for me prefectly.","info":"Output 1:\nThis output will send a **true** when Lights or Groups are turned on.\n\nOutput 2:\nThis output will send a **true** when Lights or Groups are turned on AND a **false** when Lights or Groups are turned off.\n\nOutput 3:\nThis output will send a **false** when Lights or Groups are turned off\n\nDepending on your Situation it can be useful just to have a **true** OR **false** OR **both** states in your Flow. Thats the reason i built this Subflow.\nJust use the Output that makes you happy :)\n\n-----------------------\nAll outputs are boolean \n-----------------------\n\nThe delays will reset the RBE nodes.\nYou can adjust the times for your needs, but this worked for me prefectly.","x":510,"y":220,"wires":[]},{"id":"f3271feba7c9915a","type":"switch","z":"a481bf787a7fdeab","name":"light / grouped_light ID","property":"payload.data.0.id","propertyType":"msg","rules":[{"t":"cont","v":"light_ID","vt":"env"}],"checkall":"true","repair":false,"outputs":1,"x":200,"y":580,"wires":[["db4b92b38d2635c5"]]},{"id":"ab5cf922d37164a0","type":"tab","label":"Flow 1","disabled":false,"info":"","env":[]},{"id":"993f443e1bcb551b","type":"subflow:a481bf787a7fdeab","z":"ab5cf922d37164a0","name":"","env":[{"name":"light_ID","value":"replace with discovered Lamp ID / Group RID","type":"str"}],"x":730,"y":400,"wires":[["a4277fd6dd0cab70"],["c02a99abe3837d32"],["be35a684d8f03927"]]},{"id":"3ef2d04c3e217b93","type":"link in","z":"ab5cf922d37164a0","name":"HUE IN","links":["796fb88f0dc45f4e"],"x":350,"y":240,"wires":[["993f443e1bcb551b"]],"l":true},{"id":"a4277fd6dd0cab70","type":"debug","z":"ab5cf922d37164a0","name":"debug 29","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1120,"y":320,"wires":[]},{"id":"c02a99abe3837d32","type":"debug","z":"ab5cf922d37164a0","name":"debug 30","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1120,"y":380,"wires":[]},{"id":"be35a684d8f03927","type":"debug","z":"ab5cf922d37164a0","name":"debug 31","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1120,"y":440,"wires":[]},{"id":"1661317601836c93","type":"comment","z":"ab5cf922d37164a0","name":"Insert your light ID here, don't double click on the subflow. Everything inside is set automatically. ","info":"","x":750,"y":340,"wires":[]},{"id":"e204af109c329e62","type":"comment","z":"ab5cf922d37164a0","name":"This goes to the hue event flow (like you imported, orange node flow) ","info":"","x":350,"y":200,"wires":[]},{"id":"3a65d034e2d45911","type":"change","z":"ab5cf922d37164a0","name":"Light off / 1s transition","rules":[{"t":"set","p":"endpoint","pt":"msg","to":"light","tot":"str"},{"t":"set","p":"topic","pt":"msg","to":"Replace with your discovered ID","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"{\"on\":{\"on\":false},\"dynamics\":{\"duration\":1000}}","tot":"json"},{"t":"set","p":"verb","pt":"msg","to":"put","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":720,"y":500,"wires":[["29713b8e61babb38"]]},{"id":"caa34c6a8d1a78ec","type":"change","z":"ab5cf922d37164a0","name":"Light on / 1s transition","rules":[{"t":"set","p":"endpoint","pt":"msg","to":"light","tot":"str"},{"t":"set","p":"topic","pt":"msg","to":"Replace with your discovered ID","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"{\"on\":{\"on\":true},\"dynamics\":{\"duration\":1000}}","tot":"json"},{"t":"set","p":"verb","pt":"msg","to":"put","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":720,"y":580,"wires":[["29713b8e61babb38"]]},{"id":"29713b8e61babb38","type":"link out","z":"ab5cf922d37164a0","name":"link out 17","mode":"link","links":[],"x":1085,"y":560,"wires":[]},{"id":"3e5a527716c46baf","type":"link in","z":"ab5cf922d37164a0","name":"link in 18","links":[],"x":465,"y":720,"wires":[["3a65d034e2d45911"]]},{"id":"89fe35ba8ead5ea0","type":"link in","z":"ab5cf922d37164a0","name":"link in 19","links":[],"x":555,"y":720,"wires":[["caa34c6a8d1a78ec"]]},{"id":"ab91f5f4fb06d73c","type":"inject","z":"ab5cf922d37164a0","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":460,"y":580,"wires":[["caa34c6a8d1a78ec"]]},{"id":"89e86f72a70910c5","type":"inject","z":"ab5cf922d37164a0","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":460,"y":500,"wires":[["3a65d034e2d45911"]]},{"id":"0713991d289d726c","type":"comment","z":"ab5cf922d37164a0","name":"Connect this to your wall switches of your choice ","info":"","x":520,"y":760,"wires":[]},{"id":"e84fa09ae4277d0e","type":"comment","z":"ab5cf922d37164a0","name":"This goes to the http request with the api v2 node ","info":"","x":1080,"y":600,"wires":[]},{"id":"7d4d9c9a3ce2568a","type":"comment","z":"ab5cf922d37164a0","name":"Enter IDs here","info":"","x":710,"y":540,"wires":[]},{"id":"3eb9565e1fcc9b39","type":"comment","z":"ab5cf922d37164a0","name":"All nodes need to get the same Id.. that would be a setup for one lamp. Blue returns status, yellow activates","info":"","x":730,"y":460,"wires":[]}] `


Let me know if it worked or not. 
It can be a bit confusing at the beginning, especially when you have a starter knowledge in node red (is this the case too?) :)

**Comment by hagepat** [comment]
Hi Andreas,
I will let you know this evening if it works,

Thanks for your effort.

Greetz Patrick

**Comment by hagepat** [comment]
Hi Andreas, 
wow thanks for your clear explanation, I think I understand :-)
This is the flow with the lights now
![image](https://user-images.githubusercontent.com/86970871/190226608-56cd3103-ca1f-4d87-885e-11db5a581a84.png)

This is the flow where I turn on the lights by schedule:
![image](https://user-images.githubusercontent.com/86970871/190226772-91419444-305d-43aa-9ca9-cf210ad0aa28.png)

This is the node where I switch off the light at 01:30
![image](https://user-images.githubusercontent.com/86970871/190226714-04fb6110-83eb-4af8-8441-bd7620136833.png)

I have these 2 flows now:
![image](https://user-images.githubusercontent.com/86970871/190226966-50c665eb-9719-4e87-8cfd-4057533943fe.png)

HUE API V2 is the one to configure
HUE LIghts is where I put all my lights in
![image](https://user-images.githubusercontent.com/86970871/190227112-896b216b-3574-49d8-a1be-a7e6c2dbdb73.png)

And this is how it looks with two lights:
![image](https://user-images.githubusercontent.com/86970871/190227262-4ac9e2bb-7cbb-4483-8050-5fdfa0396611.png)


Is this somewhat correct, how it should be configured?

Greetz Patrick

**Comment by andesse** [comment, marked as answer]
@hagepat when it’s working you can do it step-by-step for all your devices now, it’s a short learning process.
When you are into it you get new lamps / scenes in a couple of minutes in your flow. 
using one flow tab with link in / out is the correct way to go.

when you look into the yellow action change nodes you can also determine how the code works, and just can adjust it for your needs. 

My repository is actually just for showing how it works and providing tools and preset flows so that you can build your own setup the easiest way possible without relying on a bad maintained contrib. What you do wit it is on you;)

Greetings Andreas
